build.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+if [ -z "$ASAR" ]; then
+    ASAR=$(which asar)
+fi
+if [ -z "$ASAR" ] || [ ! -f "$ASAR" ]; then
+    echo Error: asar is required, please add it to PATH or set env var ASAR pointing to the executable >&2
+    exit 1
+fi
+ASAR=$(realpath $ASAR) # in case it has a relative path
+ROOTDIR=$(realpath .)
+
+for f in variapatches/*.asm; do
+    basename="$(basename $f .asm)"
+    if [ ! -f variapatches/ips/$basename.ips ]; then
+        echo WARNING: $f does not have a corresponding ips file in variapatches/ips !
+    fi
+done
+
+echo Building Super Metroid Multiworld Basepatch
+
+for subdir in $(ls -d vanilla romhacks/*); do
+
+    mkdir -p build/$subdir
+    python3 resources/create_dummies.py build/00.sfc build/ff.sfc
+    pushd $ROOTDIR/$subdir > /dev/null
+    $ASAR --no-title-check --symbols=wla --symbols-path=$ROOTDIR/build/$subdir/multiworld.sym main.asm $ROOTDIR/build/00.sfc
+    $ASAR --no-title-check --symbols=wla --symbols-path=$ROOTDIR/build/$subdir/multiworld.sym main.asm $ROOTDIR/build/ff.sfc
+    if [[ $? != 0 ]]; then
+        popd > /dev/null
+        exit 1
+    fi
+    popd > /dev/null
+    python3 resources/create_ips.py build/00.sfc build/ff.sfc build/$subdir/multiworld-basepatch.ips
+    echo Built: build/$subdir/multiworld-basepatch.ips
+    rm build/00.sfc build/ff.sfc
+
+    python3 resources/sym2json.py build/$subdir/multiworld.sym common/*.asm > build/$subdir/sm-basepatch-symbols.json
+
+    if [ -f $subdir/variapatches.includelist ]; then
+
+        # start with an empty patch and then merge all variapatches into it
+        echo -ne "PATCHEOF" > build/$subdir/variapatches.ips
+        for ipsref in $(cat $subdir/variapatches.includelist | grep -v "^#" | grep -v "^\/\/" | grep -v "^;"); do
+            if [ ! -f variapatches/ips/$ipsref ]; then
+                echo Error: $ipsref referenced in $subdir/variapatches.includelist: file not found at variapatches/ips/$ipsref
+                exit 1
+            fi
+            ipsfile="variapatches/ips/$ipsref"
+            basename="$(basename $ipsfile .ips)"
+            if [ ! -f variapatches/$basename.asm ]; then
+                echo WARNING: $ipsfile does not have a corresponding asm file in variapatches !
+            else
+                if [ variapatches/$basename.asm -nt $ipsfile ]; then
+                    echo WARNING: $ipsfile is older than variapatches/$basename.asm . Possible old ips patch that needs to be re-compiled - re-compiling is manual and may require xkas-plus.
+                fi
+            fi
+            echo -n " "
+            python3 resources/merge_ips.py $ipsfile build/$subdir/variapatches.ips
+        done
+        echo "Merged: build/$subdir/variapatches.ips"
+    fi
+
+done
+
+echo
+echo Done
```

Comment:
I don't know your environment but "python3" isnt recognized on mine (Win10 with build.sh called from Git Fork console). Could you change it to "python"? Its working for me that way and I think it should for you too?

Also, the patched ROM crashes as I collect Morph location. Could you tell me how you changed AP to properly apply them? I tried replacing the basepatch.ips entry in IPSPatches 'Standard' list by the 2 new ips. And I also tried to merge both together in build.sh (like I originaly did). I must be missing something.

Reply:
Thanks for checking it out lordlou! I have been sitting on some final testing for too long, but will incorporate this when I push the update! Will have a new build posted here in a few hours and will also start a PR in the AP repo at the same time so this can be more seriously tested. I'll keep the 2 PR's in sync as far as the built IPS goes.

> I don't know your environment but "python3" isnt recognized on mine (Win10 with build.sh called from Git Fork console). Could you change it to "python"? Its working for me that way and I think it should for you too?

Great suggestion, will include in the upcoming iteration. Turns out `python3` is mandated only for Unix. Thanks!

> Also, the patched ROM crashes as I collect Morph location.

Yes, sorry about that! I found the same crash somewhat recently and it's due to my having removed most of the `sm_item_graphics` table from bank $84, where AP's worlds/sm/__init__.py is still trying to write sprite-related data and ends up overwriting a PLM instruction list instead. The corrected address (now symbolic, though fun fact, it's still in bank $84 since that's where PLM palette indices have to be stored) will be in the AP PR.